python/cugraph/cugraph/structure/graph_classes.py

```diff
@@ -81,7 +81,9 @@ def __init__(self, m_graph=None, directed=False):
                 )
 
     def __getattr__(self, name):
-        if self._Impl is None:
+        # accessing `_Impl` will lead to recursion
```

Comment:
Ah, thanks, we've never tested this with `deepcopy` before so that's not too surprising.  Can you also add this as the docstring, since this code and using \_\_getattr\_\_() this way is not intuitive IMO:
```suggestion
        """
        __getattr__() is called automatically by python when an attribute does not
        exist. Since this class is attempting to hide the internal `_Impl` object,
        which is intended to contain many of the attributes needed by this class,
        __getattr__ is used to "pass through" attribute access to _Impl and make it
        appear as if the _Impl attributes are contained in this class.
        """
```